### PR TITLE
fix: SendWebRequest retries system is blocking the async operation

### LINKED ISSUE #910: WebRequestController.SendWebRequest<T> retry & return value
https://github.com/decentraland/unity-renderer/blob/master/unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/WebRequest/WebRequestController.cs#L102-L170

`WebRequestController.SendWebRequest<T> `is returning `WebRequestAsyncOperation resultOp` when called
but `resultOp` is assigned a new instance for every retry.

In the following scenario:

```
var op = WebRequestController.SendWebRequest<T>(...)
yield return op
```
if there was a retry, `op` is no longer valid for the operation and will keep waiting forever

### PR DESCRIPTION
**Pull request by sandrade-dcl**
## What does this PR change?

Fixes #910

`WebRequestController.SendWebRequest<T>` is returning `WebRequestAsyncOperation resultOp` when called
but `resultOp` is assigned with a new instance for every retry.

In the following scenario:

```
var op = WebRequestController.SendWebRequest<T>(...)
yield return op
```
If there was a retry, `op` is no longer valid for the operation and will keep waiting forever.

## Test link
https://play.decentraland.zone/index.html?renderer=urn:decentraland:off-chain:renderer-artifacts:fix/SendWebRequest-retries-system-is-blocking-the-async-operation&ENV=org

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/WebRequest/Interfaces/WebRequestAsyncOperation.cs [hunk 1]
```diff
@@ -49,6 +49,8 @@ public WebRequestAsyncOperation(UnityWebRequest webRequest)
             disposeOnCompleted = true;
         }
 
+        public void SetNewWebRequest(UnityWebRequest newRequest) { webRequest = newRequest; }
```
Comment (pbosio) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/WebRequest/Interfaces/WebRequestAsyncOperation.cs:52]: could we set this as `internal`?
If we keep it `public` it must handle internally the scenario of replacing the `webRequest` like `WebRequestController` does when calling `resultOp.Dispose()` before retrying

**Review by pbosio — APPROVED** [review]
LGTM! just a small observation:

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/WebRequest/Interfaces/WebRequestAsyncOperation.cs (hunk 1 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/WebRequest/Interfaces/WebRequestAsyncOperation.cs]: You're right! done in https://github.com/decentraland/unity-renderer/pull/949/commits/632c6957d2cbfa7703abe3fba3c0b150911bce68. Thanks!

**Review by AjimenezDCL — APPROVED** [review]
We should have a test covering this, but good job

**Review by pravusjif — APPROVED** [review]

**Pull request merged**